urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/WeiboResponse.java

```
@@ -0,0 +1,234 @@
+        String str2 = getTextContent(str, elem);
+        if (null == str2 || "".equals(str2)||"null".equals(str)) {
+            return -1;
+        } else {
+            return Long.valueOf(str2);
+        }
+    }
+
+    protected static String getString(String name, JSONObject json, boolean decode) {
+        String returnValue = null;
+            try {
+                returnValue = json.getString(name);
+                if (decode) {
+                    try {
+                        returnValue = URLDecoder.decode(returnValue, "UTF-8");
+                    } catch (UnsupportedEncodingException ignore) {
+                    }
+                }
+            } catch (JSONException ignore) {
+                // refresh_url could be missing
+            }
+        return returnValue;
+    }
+
+    protected static boolean getChildBoolean(String str, Element elem) {
+        String value = getTextContent(str, elem);
+        return Boolean.valueOf(value);
+    }
+    protected static Date getChildDate(String str, Element elem) throws WeiboException {
+        return getChildDate(str, elem, "EEE MMM d HH:mm:ss z yyyy");
+    }
+
+    protected static Date getChildDate(String str, Element elem, String format) throws WeiboException {
+        return parseDate(getChildText(str, elem),format);
+    }
+    protected static Date parseDate(String str, String format) throws WeiboException{
+        if(str==null||"".equals(str)){
+        	return null;
+        }
+    	SimpleDateFormat sdf = formatMap.get(format);
+        if (null == sdf) {
+            sdf = new SimpleDateFormat(format, Locale.ENGLISH);
+            sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+            formatMap.put(format, sdf);
+        }
+        try {
+            synchronized(sdf){
+                // SimpleDateFormat is not thread safe
+                return sdf.parse(str);
+            }
+        } catch (ParseException pe) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)